### PR TITLE
Feature/festival scrape

### PR DESCRIPTION
**Pull request by RobinStut**
We need to think about some solutions.
I added some comments, maybe we can think about this

### CODE REVIEW

File: server/server.js [hunk 1]
```diff
@@ -22,6 +24,82 @@ const upload = multer({ dest: 'server/assets/uploads/' })
 const app = express()
 const port = process.env.PORT || 3000
 
+// What is a better file/directory to place this in?
```
Comment (Mennauu) [inline server/server.js]: In `server/middleware/cron/events.js` sounds good.

File: server/server.js [hunk 2]
```diff
@@ -22,6 +24,82 @@ const upload = multer({ dest: 'server/assets/uploads/' })
...
+    'techno',
+    'trance',
+    'urban',
+  ]
+
+  puppeteer.launch().then(async browser => {
+    const page = await browser.newPage()
+    let mergedData = []
+
+    // this has to be an oldschool for loop, because forEach, map and
```
Comment (Mennauu) [inline server/server.js]: and? XD

Comment (RobinStut) [reply server/server.js]: "map, forEach, reduce, etc, does not wait for the asynchronous operation within them."
https://stackoverflow.com/a/46297546

Comment (Mennauu) [reply server/server.js]: You can write it like this?

for (const [i, genre] of musicGenres.entries()) {
   
}

File: server/server.js [hunk 3]
```diff
@@ -22,6 +24,12 @@ const upload = multer({ dest: 'server/assets/uploads/' })
 const app = express()
 const port = process.env.PORT || 3000
 
+// Job starts when clock hits the 0 hour, the 0 minute
```
Comment (Mennauu) [inline server/server.js:27]: Can't you also add this to to the events.js file?

Afterwards you can change the import
 ```JavaScript 
import getFestivalData from './middleware/cron/events.js'
``` 
to 
 ```JavaScript 
import './middleware/cron/events.js'
```
In order to start the cron job.

File: server/middleware/cron/events.js [hunk 4]
```diff
@@ -0,0 +1,72 @@
+import puppeteer from 'puppeteer'
+
+// What is a better file/directory to place this in?
+export default async function getFestivalData() {
```
Comment (Mennauu) [inline server/middleware/cron/events.js:4]: Might want to include `try ... catch` block

File: server/middleware/cron/events.js [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+    'house',
+    'pop',
+    'rb',
+    'rock',
+    'techno',
+    'trance',
+    'urban',
+  ]
+
+  puppeteer.launch().then(async browser => {
```
Comment (Mennauu) [inline server/middleware/cron/events.js:20]: Might want to include `try ... catch` block

File: server/middleware/cron/events.js [hunk 6]
```diff
@@ -1,13 +1,26 @@
...
+
+  if (!isRecentlyUpdated) {
+    setFestivalData()
+  }
 })
+
 export default job.start()
 
-async function getFestivalData() {
+async function setFestivalData() {
```
Comment (chelseadoeleman) [inline server/middleware/cron/events.js:23]: Really just a minor detail, but I see you're using normal functions as well as arrow functions. Might want to use one or the other consistently 😊. Everything else looks good!

Comment (RobinStut) [reply server/middleware/cron/events.js]: Changed it! Thanks!

**Comment by Mennauu** [comment]
Approved!

**Pull request merged**